StyleCop.Analyzers/StyleCop.Analyzers.CodeFixes/ReadabilityRules/SA1133CodeFixProvider.cs

```diff
@@ -80,10 +91,41 @@ public override Task RegisterCodeFixesAsync(CodeFixContext context)
                 newAttributeLists.Add(newAttributeList);
             }
 
-            var newSyntaxRoot = syntaxRoot.ReplaceNode(attributeList, newAttributeLists);
-            var newDocument = document.WithSyntaxRoot(newSyntaxRoot.WithoutFormatting());
+            return newAttributeLists;
+        }
 
-            return newDocument;
+        private class FixAll : DocumentBasedFixAllProvider
+        {
+            public static FixAllProvider Instance { get; } =
+                new FixAll();
+
+            protected override string CodeActionTitle =>
+                ReadabilityResources.SA1133CodeFix;
+
+            protected override async Task<SyntaxNode> FixAllInDocumentAsync(FixAllContext fixAllContext, Document document)
+            {
+                var diagnostics = await fixAllContext.GetDocumentDiagnosticsAsync(document).ConfigureAwait(false);
+                if (diagnostics.IsEmpty)
+                {
+                    return null;
+                }
+
+                var indentationOptions = IndentationOptions.FromDocument(document);
+                var syntaxRoot = await document.GetSyntaxRootAsync(fixAllContext.CancellationToken).ConfigureAwait(false);
+
+                var nodes = diagnostics.Select(diagnostic => syntaxRoot.FindNode(diagnostic.Location.SourceSpan, getInnermostNodeForTie: true).FirstAncestorOrSelf<AttributeListSyntax>());
+
+                var newRoot = syntaxRoot.TrackNodes(nodes);
+
+                foreach (var attributeList in nodes)
+                {
+                    var indentationSteps = IndentationHelper.GetIndentationSteps(indentationOptions, attributeList);
+                    var indentationTrivia = IndentationHelper.GenerateWhitespaceTrivia(indentationOptions, indentationSteps);
+                    newRoot = newRoot.ReplaceNode(newRoot.GetCurrentNode(attributeList), GetNewAttributeList(attributeList, indentationTrivia));
+                }
```

Comment:
Yup, I forgot about that. I just had a look at the `SyntaxRewriter` as well and it also suffers from the same problem. 
It could still be done using the parent node of the `AttributeListSyntax`, but that would be a large amount of work and I'm not sure if the performance gain is significant enough for that.